Draft of Move Update and Sleep

SandytheLizard opened the pull request.

SandytheLizard committed:
Small Change, Makes Stats End after Combat

SandytheLizard committed:
Draft of Move Update and Sleep

Trying to fix code for sleep status

bitcraft commented on combat.py:
```diff
@@ -0,0 +1,900 @@
+#
```
not sure what this file is here.  please update the pr to show a proper diff

bitcraft commented on mods/tuxemon/l18n/en_US/LC_MESSAGES/base.po:
```diff
@@ -160,8 +160,14 @@ msgstr "Go {name}!"
+msgstr "{target} is still sleeping."
+
+msgid "combat_woke_up"
+msgstr "{target} woke up"
```
punctuation

bitcraft commented on mods/tuxemon/l18n/en_US/LC_MESSAGES/base.po:
```diff
@@ -193,6 +199,8 @@ msgstr "{user} rose higher into the air."
 msgstr "{user} has gotten amnesia."
 
+msgid "combat_state_lock_on"
+msgstr "{target}, locked on \n, your next attack has 100% accuracy"
```
punctuation

bitcraft commented on tuxemon/technique.py:
```diff
@@ -220,7 +223,7 @@ def keep_old_stats(self):
         return self.old_stats_data
     def use(self, user: Monster, target: Monster) -> TechniqueResult:
-        
+
```
please avoid changing whitespace unless it is for PEP8

bitcraft commented on tuxemon/technique.py:
```diff
@@ -260,7 +263,10 @@ def use(self, user: Monster, target: Monster) -> TechniqueResult:
         # TODO: handle conflicting values from multiple technique actions
         # TODO: for example, how to handle one saying success, and another not?
-       
+        print(target.sleep)
```
print

bitcraft commented on tuxemon/technique.py:
```diff
@@ -272,12 +278,15 @@ def use(self, user: Monster, target: Monster) -> TechniqueResult:
-                result = self.apply_status("status_overfeed", target)
+            elif effect == "sleep":
+                result = self.apply_sleep(user, target)
+                print("b")
```
print

bitcraft commented on tuxemon/technique.py:
```diff
@@ -260,7 +263,10 @@ def use(self, user: Monster, target: Monster) -> TechniqueResult:
         # TODO: for example, how to handle one saying success, and another not?
-       
+        print(target.sleep)
+        if target.sleep > 0:
```
probably not likely to happen, but if a technique has other effects (poison?), and the target is sleeping, then the other effects will never be processed.  if poison was applied separately, it would still take damage, i believe.

bitcraft reviewed (changes requested):
looks good so far.  need to get the combat file fixed, but everything else looks fine.

SandytheLizard committed:
Things

1. Redid about 30% of the moves
2.Added a cli command called random_fight. this generates a random 6v6 battle. This command will crash if you have tuxemon in your party upon execution or if it picks a tuxemon without a valid shape.

3. Made it so stats revert after a battle.

SandytheLizard committed:
Merge branch 'development' of https://github.com/Tuxemon/Tuxemon into development

SandytheLizard committed:
things

SandytheLizard committed:
Merge branch 'development' of https://github.com/SandytheLizard/Tuxemon into development
